### DISCUSSION TITLE
run from source

### DISCUSSION THREAD

**Discussion by Timoyoungster** [Cloudef/bemenu#408] [Q&A] [answered]
Hello,

I tried to build and run bemenu from source but always got the error that the lib files couldn't be found.
With the environment variables from the readme explicitly set to the correct paths it works, but I can't seem to make it work with hyprland. (even though I set the vars in my hypr.conf, bemenu doesn't launch when called by hyprland)

I also tried installing it to /usr instead of /usr/local which removed the lib.so not found error, but it still didn't display. When run directly from the command line it just silently exited.

My main reason for doing so is, because I would like to submit/run on C-y in addition to return and would have therefore changed the code myself accordingly, but can't even get the original source to run from hyprland.

What is the difference between the pacman installed package and the 'make install'ed version such that it can't find the lib.so files even though I set the env variable in my hypr.conf file?

Thanks in advance!

**Comment by Cloudef** [comment, marked as answer]
If you want to install from source without having to use the env vars when running bemenu, you have to make sure you set `PREFIX=/usr` to correct location, and the bemenu renderers must be found from `$PREFIX/lib/bemenu/...`
https://github.com/Cloudef/bemenu/blob/5bc8a6caaa74f2255fbf4ef8185be50c6e3e6fbe/lib/library.c#L130

So in your case `PREFIX=/usr/local` should work. (Which should be default)

Note that bemenu still looks for `libbemenu.so` from a _system library search path_ so you either have to configure alternative search path in `/etc/ld.so.conf` or set the `LD_LIBRARY_PATH`.

Alternatively if you have nix or flatpak, you can build bemenu with those.
If hyrpland has trouble passing the env vars to bemenu, you might want to try write a wrapper script instead which sets those vars and calls your compiled bemenu.

**Reply by Timoyoungster** [reply]
Thanks! The ld.so.conf did the trick.
